braintrust/trace/internal/middleware.go

```diff
@@ -33,29 +33,47 @@ func Middleware(router TracerRouter) func(*http.Request, NextMiddleware) (*http.
 		log.Debugf("Middleware: %s %s", req.Method, req.URL.Path)
 		start := time.Now()
 
-		// Intercept the request body so we can parse it and still pass it along.
+		// Determine which tracer to use first
+		var reqTracer MiddlewareTracer
+		if req.URL != nil {
+			reqTracer = router(req.URL.Path)
```

Comment:
i don't love the name of the router function her. a router returns a middleware tracer. why not call it 

`mt := getMiddlewareTracer`